capstone_project_EEG_detection.ipynb

```diff
@@ -573,214 +598,98 @@
     "print(\"\\n\\nAmplitude Range Of Interest:\")\n",
     "display(test_gpac_data['hf'])\n",
     "\n",
-    "print(f\"\\n\\nNumber of Datapoints Per GPAC Image: {len(test_gpac_data['data']['fv'][0])}\")\n"
+    "print(f\"\\n\\nNumber of Datapoints Per GPAC Image: {len(test_gpac_data['data']['fv'][0])}\")\n",
+    "\n",
+    "\n",
+    "# Eye blinking data\n",
+    "print(\"\\n\\n\\n---------- EYE BLINKING DATA FORMAT ----------\")\n",
+    "\n",
+    "# Note that for the eye blinking data, multiple windows from different sessions live together in the same\n",
+    "# data structure. Each of these windows is 4s in length. As a result, time does not increase in each\n",
+    "# successful row in the data structure indefinitely, but rather increases from 0 to (4 - 1/256) and then\n",
+    "# will reset to 0 for the next row.\n",
+    "print(\"\\n\\nData For Training:\")\n",
+    "display(test_eye_blinking_data['data'])\n",
+    "\n",
+    "print(\"\\n\\nElectrode Names:\")\n",
+    "display(test_eye_blinking_data['electrodes'])\n",
+    "\n",
+    "print(f\"\\n\\nNumber of Channels Per Sample: {len(test_eye_blinking_data['data']['samples'][0])}\")\n",
+    "\n",
+    "\n",
+    "# Hand movement data\n",
+    "print(\"\\n\\n\\n---------- HAND MOVEMENT DATA FORMAT ----------\")\n",
+    "\n",
+    "print(\"\\n\\nData For Training:\")\n",
+    "display(test_hand_movement_data['data'])\n",
+    "\n",
+    "print(f\"\\n\\nNumber of Channels Per Sample: {len(test_hand_movement_data['data']['samples'][0])}\")\n",
+    "\n",
+    "\n",
+    "# Resting state data\n",
+    "print(\"\\n\\n\\n---------- BASELINE DATA FORMAT ----------\")\n",
+    "\n",
+    "print(\"\\n\\nData For Training:\")\n",
+    "display(test_baseline_data['data'])\n",
+    "\n",
+    "print(\"\\n\\nElectrode Names:\")\n",
+    "display(test_baseline_data['electrodes'])\n",
+    "\n",
+    "print(f\"\\n\\nNumber of Channels Per Sample: {len(test_baseline_data['data']['samples'][0])}\")\n",
+    "\n",
+    "\n",
+    "# OSF data\n",
```

Comment:
So this just loads the file for all the datasets, and displays the format, correct? no homogenization is performed from what I can see. I think it would be important to not only display the size of the dataframe but also the size of the arrays it contains. Beyond just the changes that need to be made to balance the datasets, it is important for the datasets to be able to be processed into GPAC uniformly. As the one managing the data processing, please ensure that the arrays the the GPAC transform receives are uniform; @ChrisDryden pls weigh in on this, but I am under the impression that the arrays passed in (regardless of the window they represent) have to be the same length. We can have multiple arrays per window, but they have to be consistent. Also, did you check how many arrays there are per row?

Reply:
As per our conversation, each row is the number of channels in a given sample. That data is printed, on line 638.